aiida/cmdline/commands/cmd_import.py

```diff
@@ -127,15 +136,64 @@ def cmd_import(archives, webpages, group, extras_mode_existing, extras_mode_new,
                 extras_mode_existing=ExtrasImportCode[extras_mode_existing].value,
                 extras_mode_new=extras_mode_new,
                 comment_mode=comment_mode)
-        except exceptions.IncompatibleArchiveVersionError as exception:
-            echo.echo_critical('{} cannot be imported: {}'.format(archive, exception))
+        except exceptions.IncompatibleArchiveVersionError:
+            echo.echo_warning('{} cannot be imported.'.format(archive))
```

Comment:
Let's change this now to a more explanative message about the version conflict.

Reply:
It can be either a version conflict of the export file or of AiiDA.

Maybe "{} cannot be imported due to a version conflict" - and then we do not mention what the version belongs to? :)

Reply:
What I meant is: it *can* be imported, right? E.g. replace the message with something like:

`'{}' has export version '{}' while AiiDA v'{}' uses export version '{}'`

Reply:
At that point in time it cannot. But I understand what you mean.

I'd rather not have to retrieve the version from the file, since this requires a bit too much code simply for an error message. But it could be something like: `{} has an old export version, while AiiDA v{} uses export version {}'. That will be easy. However, it may still not be a sufficient error message.

What I mean above is that if an `IncompatibleArchiveVersionError` is raised it can mean two different things:

1. the export version of the archive is older than the current export version of the installed version of AiiDA.
2. the export version of the archive is newer than the currect export version of the installed version of AiiDA.

For the 1st point the archive should be migrated, for the 2nd one needs to install a newer version of AiiDA to import it.

Reply:
I wrote an `Archive` class that can be used as a context manager to very simply get access to some of the metadata. I used it in `verdi export inspect` and you can take inspiration from there. To get the version all you need something like this:
```
from aiida.common.archive import Archive

with Archive(archive) as archive_object:
        echo.echo(archive_object.version_format)
```

Reply:
Thanks @sphuber. Will take a look at it.
If I am using this anyway, I might as well provide the user with "intelligent" error messages.

Reply:
I have realized I can simply use the exception message thrown by `importexport.py`. However, to make it "intelligent" I would have to edit the exception message.
What is better: Edit "external" exception message or implement the same logic used "externally" to create the, essentially same, exception message "locally"?

Reply:
I have gone the way of the less involved: edit "external" exception message. Basically, I am just keeping the first line that mentions the export versions.